pkgs/os/os.sh

```diff
@@ -59,27 +63,61 @@ _refresh() {
 
 _switch() {
     echo "${1}" > "${CURRENT_FILE}"
+    _refresh
+}
+
+_check_zone() {
+  ZONE=${1}
+  if ! jq -e -r --arg ZONE "${ZONE}" '.[] | select(.providers.fe.domain.name == $ZONE) | .infra_zone_name' "${CAASCAD_ZONES_FILE}" &>/dev/null && 
+    ! jq -e -r --arg ZONE "${ZONE}" '.[$ZONE].infra_zone_name' "${CAASCAD_ZONES_FILE}" &>/dev/null; then
+    echo "Zone or domain name ${ZONE} not found" && exit 1;
+  fi
+}
+
+_get_vault_fqdn() {
+  ZONE="${1}"
+  if [[ $ZONE_NAME =~ ^OCB000.* ]]; then
+      INFRA_ZONE_NAME="$(jq -e -r --arg ZONE "${ZONE}" '.[] | select(.providers.fe.domain.name == $ZONE) | .infra_zone_name' "${CAASCAD_ZONES_FILE}" )"
+      DOMAIN_NAME="$(jq -r --arg ZONE "${INFRA_ZONE_NAME}" '.|to_entries[] | select(.key == $ZONE) | .value.domain_name' "${CAASCAD_ZONES_FILE}" )"
+  else
+      INFRA_ZONE_NAME="$(jq -e -r --arg ZONE "${ZONE}" '.[$ZONE].infra_zone_name' "${CAASCAD_ZONES_FILE}")"
+      DOMAIN_NAME="$(jq -r --arg ZONE "${ZONE}" '.[$ZONE].domain_name' "${CAASCAD_ZONES_FILE}")"
+  fi
+  echo "https://vault.${INFRA_ZONE_NAME}.${DOMAIN_NAME}"
+}
+
+_get_current_zone() {
+  ZONE=$(cat "${CURRENT_FILE}")
+  if [ -z "${ZONE}" ]; then
+    echo "Zone name empty. Use os switch first." >&2
+    exit 1
+  fi
+  echo "${ZONE}"
 }
 
 _parse() {
     if [[ "$#" -eq 0 ]]; then
-        cat "${CURRENT_FILE}";
+        _get_current_zone;
```

Comment:
finding the help is kind of hard, couldn't we display help by default ? 
(don't know if lots of stuff rely on the current behaviour ?)

Reply:
done